Code-the-Dream-School/dd-prac-team1-back
# setup sendgrid and implement email contact feature

## One Line Description
To implement the email contact feature, set up SendGrid.## Requirements

_What this code should do_
Users can send an email by using the Contact Us form.

## Test Steps

_Instructions on how to test the changes_

You can check it using either Postman or Thunder clients in VS. With Thunder Client, you send HTTP requests to the endpoints defined in the `sendMail_route` and examine the response to make sure it functions as expected.

/- Guide:
/- Register on SendGrid by visiting the official website https://sendgrid.com/
/- Obtain the required API keys and add them to the .env file.
/-After submitting a pull request, run "npm install" to install the necessary dependencies.
/-For sending email, set the request method to POST and set the URL to your backend API's sendmail endpoint (e.g., localhost:3000/api/v1/sendmail).
/- set the necessary headers (e.g., Content-Type: application/json) to indicate that we are sending JSON data in the request body.
In the request body, provide the required fields (e.g., name, email, message) with sample data to test the endpoints. Need to make sure to use valid data that adheres to any validation rules you have in place.
/- click on the "Send Request" for each request to send the HTTP request to backend server.
/- check the response received for the requests, any errors, or status code.
/- For user sending email, the response should have a successful status code (e.g., 200) and a JSON object with message:  "Email sent successfully"

## Notes
Implement the emailRateLimiter middleware to ensure a user can only send a maximum of 5 emails within a 15-minute window.

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed
